guides/recipes/anti-patterns.md

```diff
@@ -0,0 +1,843 @@
+- Increased backup and maintenance costs
+- Potential scaling issues as job volume grows
+- Dependency on a table designed for transient data
+
+### Example
+
+Insufficient job pruning configuration:
+
+```elixir
+config :my_app, Oban,
+  engine: Oban.Pro.Engines.Smart,
+  repo: MyApp.Repo,
+  plugins: [
+    # No pruner configured, or pruner with excessive retention
+    {Oban.Plugins.Pruner, max_age: 60 * 60 * 24 * 365}, # 1 year
+    Oban.Pro.Plugins.DynamicLifeline
+  ],
+  queues: [default: 10, external_service: 5]
+```
+
+Code that improperly relies on long-term job persistence:
+
+```elixir
+defmodule MyApp.Reports do
+  def generate_monthly_report(year, month) do
+    # Problematic: Relies on jobs to remain in the database indefinitely
+    jobs =
+      Oban.Job
+      |> where([j], j.worker == "MyApp.Workers.InvoiceProcessor")
+      |> where([j], j.state == "completed")
+      |> where([j], fragment("?->>'year' = ?", j.args, ^to_string(year)))
+      |> where([j], fragment("?->>'month' = ?", j.args, ^to_string(month)))
+      |> MyApp.Repo.all()
+
+    # Process jobs for reporting
+    process_jobs_for_report(jobs)
+  end
+end
+```
+
+### Refactoring
+
+Use appropriate pruning configuration with Oban Pro's full range of options:
+
+```elixir
+config :my_app, Oban,
+  engine: Oban.Pro.Engines.Smart,
+  repo: MyApp.Repo,
+  plugins: [
+    # Comprehensive pruning strategy with different retention periods
+    {Oban.Plugins.Pruner, 
+      # Standard states can be pruned fairly quickly
+      max_age: 60 * 60 * 24 * 7,            # 1 week for completed/cancelled
+      # Keep failures longer for investigation
+      max_age_failure: 60 * 60 * 24 * 30,   # 30 days for failed jobs
+      # Limit total jobs regardless of age
+      limit: 250_000                        # Cap total jobs
+    },
+    Oban.Pro.Plugins.DynamicLifeline
+  ],
+  queues: [default: 10, external_service: 5]
+```
+
+Create a dedicated schema for job results that need longer retention:
+
+```elixir
+defmodule MyApp.Schema.JobResult do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "job_results" do
+    field :job_type, :string
+    field :parameters, :map
+    field :result, :map
+    field :year, :integer
+    field :month, :integer
+    field :processed_at, :utc_datetime
+
+    timestamps()
+  end
+
+  def changeset(record, attrs) do
+    record
+    |> cast(attrs, [:job_type, :parameters, :result, :year, :month, :processed_at])
+    |> validate_required([:job_type, :processed_at])
+  end
+end
+
+defmodule MyApp.JobResults do
+  import Ecto.Query
+  alias MyApp.Repo
+  alias MyApp.Schema.JobResult
+
+  def create(attrs) do
+    %JobResult{}
+    |> JobResult.changeset(attrs)
+    |> Repo.insert()
+  end
+
+  def get_by_period(job_type, year, month) do
+    JobResult
+    |> where([r], r.job_type == ^job_type)
+    |> where([r], r.year == ^year)
+    |> where([r], r.month == ^month)
+    |> Repo.all()
+  end
+end
```

Comment:
Since the example is weak, the refactor also seems weak

Reply:
Using a job result temporarily is fine, which is why there's a `recorded` option in the first place. If the result is important in the long term then it should be written to a proper database entity, and certinly not a `JobResult`.